versioned_hdf5/api.py

```diff
@@ -320,6 +322,29 @@ def write(self, mspace, fspace, arr_obj, mtype=None, dxpl=None):
 
         return data_dict
 
+    def read(self, mspace, fspace, arr_obj, mtype=None, dxpl=None):
```

Comment:
Please add a docstring

Reply:
I think it will be simpler for me to do a pass later documenting all the internal stuff. I should also probably move a lot of these things out of api.py since they aren't actually part of the public API.